graphblas_algorithms/nxapi/shortest_paths/weighted.py

```diff
@@ -0,0 +1,36 @@
+from graphblas_algorithms import algorithms
+from graphblas_algorithms.classes.digraph import to_graph
+
+from ..exception import NetworkXUnbounded, NodeNotFound
+
+__all__ = [
+    "all_pairs_bellman_ford_path_length",
+    "single_source_bellman_ford_path_length",
+]
+
+
+def all_pairs_bellman_ford_path_length(G, weight="weight"):
+    # TODO: what if weight is a function?
+    # How should we implement and call `algorithms.all_pairs_bellman_ford_path_length`?
+    # Should we compute in chunks to expose more parallelism?
```

Comment:
I'm not sure what the best API for a backend implementation (i.e., to have in `graphblas_algorithms.algorithms`). Returning a `Matrix` seems risky.

Maybe `bellman_ford_path_lengths(G, nodes=None)`, which will return a `Matrix` with distances for the specified nodes (default to all nodes). Then, `nxapi.all_pairs_bellman_ford_path_length` can compute in _batches_ to get increased parallelism (assuming doing so is actually faster).